### PR TITLE
Onboarding applicationWhitelistings to a new stable API version - 2020-01-01

### PR DESCRIPTION
**Pull request by erelh**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by erelh** [comment]
Please do not merge yet.

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/applicationWhitelistings.json [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+            "$ref": "../../../common/v1/types.json#/parameters/AscLocation"
+          },
+          {
+            "$ref": "#/parameters/GroupName"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "#/parameters/AppWhitelistingGroupDataPutDataBody"
```
Comment (ravbhatnagar) [inline specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/applicationWhitelistings.json]: the request body and response body must be same. it helps in get put pipeline. Customer GETs the resource, makes some changes and then PUts using the same model that he got using GET. If models are different, it becomes harder to use. 
Any properties which cant be set in request but come back during response can be marked as readOnly:true.

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/applicationWhitelistings.json [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
```
Comment (ravbhatnagar) [inline specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/applicationWhitelistings.json]: consider supporting a PATCH to allow the users to update a subset of properties without having to provide the entire payload.

**Review by ravbhatnagar — COMMENTED** [review]
@erelh few comments. please take a look

**Review by chlahav — APPROVED** [review]

File: specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/applicationWhitelistings.json (hunk 2 above)
Comment (erelh) [reply specification/security/resource-manager/Microsoft.Security/stable/2020-01-01/applicationWhitelistings.json]: > consider supporting a PATCH to allow the users to update a subset of properties without having to provide the entire payload.

We added this to our backlog, thank you

**Comment by ravbhatnagar** [comment]
approving from ARM side. Patch support is added to the backlog and will be done later. But these may be proxy types anyway.

**Comment by erelh** [comment]
Please do not merge yet.

**Comment by MatanShabtay** [comment]
@raych1 can you please remove the DoNotMerge flag and complete the PR? thanks.

**Comment by raych1** [comment]
@MatanShabtay , can you resolve the conflicts?

**Comment by erelh** [comment]
@raych1 it seems there's an issue with the validation:
the same code that was already approved is now failing.
one of the parameters is a subscription id that should pass the guid regex - and for some reason the validation removes the "-" from the string and so the validation fails.

**Comment by raych1** [comment]
CI checks status isn't updated. Closed and reopen PR to trigger CI.

**Comment by raych1** [comment]
@jianyexi , can you take a look at the lint failures?

**Comment by jianyexi** [comment]
> @jianyexi , can you take a look at the lint failures?

It caused by the conflict when autorest composing all the input file in tags "package-composite-v3", it's not a lint error, but an  autorest error.
the values of `$.paths["/subscriptions/{subscriptionId}/providers/Microsoft.Security/applicationWhitelistings"].get["x-ms-examples"]["Gets a list of application control VM/server groups for the subscription"].parameters["api-version"]`  are not consistent in the input files

**Comment by raych1** [comment]
@erelh , let me know if this PR is ready to merge.

**Comment by erelh** [comment]
ready to merge, thanks

**Pull request merged**